packages/ts/react-form/test/useForm.spec.tsx

```diff
@@ -321,5 +321,36 @@ describe('@vaadin/hilla-react-form', () => {
       await user.click(loadFormBtn);
       await expect(findByTestId('contracts')).to.eventually.have.nested.property('selectedOptions.0.value', '202');
     });
+
+    it('runs validators on change event', async () => {
+      function ProjectForm() {
+        const { addValidator, field, model } = useForm(EntityModel);
+        const [count, setCount] = useState(0);
+
+        useEffect(() => {
+          addValidator({
+            message: 'ID expected',
+            validate: (entity) => {
+              setCount(count + 1);
+              return !entity.projectId;
```

Comment:
Regarding spy usage: It's possible to define a spy outside of the component and call the spy here. That requires adding a timeout before the assertion though, it looks like there is some async processing in the binder. Not sure if that makes things easier.

Reply:
Thanks. I guess we can keep the current approach as it's not very complicated anyway.